src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-form/rgw-storage-class-form.component.ts

```diff
@@ -130,23 +149,30 @@ export class RgwStorageClassFormComponent extends CdForm implements OnInit {
       zonegroup: new FormControl(this.selectedZoneGroup, {
         validators: [Validators.required]
       }),
-      region: new FormControl('', {
-        validators: [Validators.required]
-      }),
+      region: new FormControl('', [
+        CdValidators.composeIf({ storageClassType: 'cloud-s3' }, [Validators.required])
+      ]),
       placement_target: new FormControl('', {
         validators: [Validators.required]
       }),
-      endpoint: new FormControl(null, {
-        validators: [Validators.required]
-      }),
-      access_key: new FormControl(null, Validators.required),
-      secret_key: new FormControl(null, Validators.required),
-      target_path: new FormControl('', {
-        validators: [Validators.required]
-      }),
+      endpoint: new FormControl(null, [
+        CdValidators.composeIf({ storageClassType: 'cloud-s3' }, [Validators.required])
```

Comment:
```suggestion
        CdValidators.composeIf({ storageClassType: TIER_TYPE.CLOUD_TIER}, [Validators.required])
```
also at other places